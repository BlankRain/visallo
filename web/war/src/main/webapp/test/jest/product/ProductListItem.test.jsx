import React from 'react'
import renderer from 'react-test-renderer'
import ProductListItem from 'product/ProductListItem'

it('should have a product list', () => {
    const props = {
        editable: false,
        selected: 'a',
        registry: {
            'org.visallo.workproduct': []
        },
        product: {
            id: 'a',
            title: 'a',
            workspaceId: 'a',
            kind: 'g'
        }
    }
    const component = renderer.create(<ProductListItem {...props} />, {
        createNodeMock(element) {
            if (element.type === 'input') {
                return { select: () => {}, focus: () => {} };
            } else if (element.type === 'div') {
                return { addEventListener: () => {} }
            }
            return null;
        }
    })
    expect(component.toJSON()).toMatchSnapshot()
    //const list = shallow(<ProductList {...props} />);
    //expect(list)
})
