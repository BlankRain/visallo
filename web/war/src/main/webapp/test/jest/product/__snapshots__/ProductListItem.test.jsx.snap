exports[`test should have a product list 1`] = `
<div
  className=""
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    className="products-list-item active"
    onClick={[Function]}
    onMouseLeave={[Function]}
    title="a">
    <div
      className="buttons" />
    <div
      className="no-preview"
      style={Object {}} />
    <div
      className="content">
      <h1>
        a
      </h1>
      <h2>
        g.name
      </h2>
    </div>
  </div>
</div>
`;
